dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/utils/ProgramBuilder.java

```diff
@@ -156,7 +158,7 @@ public Location getOrErrorLocation(String name){
         if(locations.containsKey(name)){
             return locations.get(name);
         }
-        throw new ParsingException("Location " + name + " has not been initialised");
+        throw new UninitialisedVariableException("Location " + name + " has not been initialised");
```

Comment:
Seeing this usage, I think `UninitialisedVariableException` should not have inherited `MalformedProgramException` unless these methods are **only** used during program construction.
Even then, `IllegalStateException` seems to be the most fitting exception.